src/services/storage/index.ts

```diff
@@ -5,19 +5,31 @@ import { getObject } from 'sequelize-store'
 import Eth from 'web3-eth'
 import { EventData } from 'web3-eth-contract'
 import { AbiItem } from 'web3-utils'
+
 import { ethFactory } from '../../blockchain'
 import { getEventsEmitterForService, isServiceInitialized } from '../../blockchain/utils'
 import { Application, CachedService, ServiceAddresses } from '../../definitions'
 import { loggingFactory } from '../../logger'
 import { errorHandler, waitForReadyApp } from '../../utils'
-import StorageOffer from '../storage/models/offer.model'
 import Agreement from './models/agreement.model'
 import Offer from './models/offer.model'
 import BillingPlan from './models/price.model'
-import hooks from './storage.hooks'
+import offerHooks from './hooks/offers.hooks'
+import agreementHooks from './hooks/agreements.hooks'
 import eventProcessor from './storage.processor'
+import storageChannels from './storage.channels'
 
 export class OfferService extends Service {
+  emit?: Function
+}
+
+export class AgreementService extends Service {
+  emit?: Function
+}
+
+export interface StorageServices {
+  agreementService: AgreementService & { emit?: Function }
```

Comment:
is this `AgreementService & { emit?: Function }` necessary if we have the emit in the class definition? Also wouldn't it be better to have something like
```
export interface StorageService extends Service {
  emit?: Function
}

export interface StorageServices {
  agreementService: StorageService,
  offerService: StorageService
}
```
alternatively (should they ever differ)
```
class AgreementService implements StorageService { ... }
...
```?

Reply:
but I guess it's not a biggie anyway so feel free to ignore it. It's just that I would personally find that clearer.

Reply:
I copy-pasted from RNS (a.k.a yours) part 😝  But yeah agree, if it will work I am down.

Reply:
:D